engine/src/test/java/org/operaton/bpm/engine/impl/jobexecutor/setInitialRetries/JobDeclarationRetriesTest.java

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.operaton.bpm.engine.impl.jobexecutor.setInitialRetries;
+
+import org.operaton.bpm.engine.ManagementService;
+import org.operaton.bpm.engine.RuntimeService;
+import org.operaton.bpm.engine.runtime.Job;
+import org.operaton.bpm.engine.test.jobexecutor.FailingDelegate;
+import org.operaton.bpm.engine.test.junit5.ProcessEngineExtension;
+import org.operaton.bpm.engine.test.util.ProcessEngineTestRule;
+import org.operaton.bpm.model.bpmn.Bpmn;
+import org.operaton.bpm.model.bpmn.BpmnModelInstance;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class JobDeclarationRetriesTest {
+
+  @RegisterExtension
+  static ProcessEngineExtension engineRule = ProcessEngineExtension.builder().build();
+
+  @RegisterExtension
+  ProcessEngineTestRule testRule = new ProcessEngineTestRule(engineRule);
+
+  private ManagementService managementService;
+  private RuntimeService runtimeService;
+
+  @BeforeEach
+  void init() {
+    this.managementService = engineRule.getManagementService();
+    this.runtimeService = engineRule.getRuntimeService();
+  }
+
+  @Test
+  void testRetryTimeCycleWithZeroRetriesAndFailure() {
+    // given
+    
+    String retryInterval = "R0/PT5M";
+    String processDefinitionName = "testRetryTimeCycleWithZeroRetriesAndFailure";
+    BpmnModelInstance bpmnModelInstance = getBpmnModelInstance(processDefinitionName, retryInterval);
+    testRule.deploy(bpmnModelInstance);
+
+    // when
+    String processInstanceId = runtimeService.startProcessInstanceByKey(processDefinitionName).getId();
+    Job job = managementService.createJobQuery().processInstanceId(processInstanceId).singleResult();
+
+    // then
+    assertThat(job.getRetries()).isEqualTo(1);
+
+    // when
+    try {
+      managementService.executeJob(job.getId());
+    } catch (Exception e) {
+      // ignore
+    }
+    job = managementService.createJobQuery().processInstanceId(processInstanceId).singleResult();
+
+    // then
+    assertThat(job.getRetries()).isZero();
+  }
+
+  @Test
+  void testRetryTimeCycleWithFailure() {
+    // given
+    
+    String retryInterval = "R5/PT5M";
+    String processDefinitionName = "testRetryTimeCycleWithFailure";
+    BpmnModelInstance bpmnModelInstance = getBpmnModelInstance(processDefinitionName, retryInterval);
+    testRule.deploy(bpmnModelInstance);
+
+    // when
+    String processInstanceId = runtimeService.startProcessInstanceByKey(processDefinitionName).getId();
+    Job job = managementService.createJobQuery().processInstanceId(processInstanceId).singleResult();
+    assertThat(job.getRetries()).isEqualTo(5);
+
+    // when
+    try {
+      managementService.executeJob(job.getId());
```

Comment:
use `assertThatThrownBy()` here
`job.getId()` must be declared in the given part as variable